src/hooks/useOnOutsideClick.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {useEffect, useCallback} from 'react'
+
+export type TouchOrMouseEvent = MouseEvent | TouchEvent
+
+export type UseOnOutsideClickProps = {
+  containerRef: React.RefObject<HTMLDivElement>
+  ignoreClickRefs?: React.RefObject<HTMLElement> []
+  onClickOutside: (e: TouchOrMouseEvent) => void
+}
+
+type ShouldCallClickHandlerProps = {
+  ignoreClickRefs?: React.RefObject<HTMLElement> []
+  containerRef: React.RefObject<HTMLDivElement>
+  e: TouchOrMouseEvent
+}
+
+const shouldCallClickHandler = ({ignoreClickRefs, containerRef, e}: ShouldCallClickHandlerProps) : boolean => {
+  let shouldCallHandler = true
+
+  // don't call click handler if the mouse event was triggered by an auxiliary button (right click/wheel button/etc)
+  if (e instanceof MouseEvent && e.button > 0) {
+    shouldCallHandler = false
+  }
+
+  // don't call handler if the click happened inside of the container
+  if (containerRef && 'current' in containerRef && containerRef.current?.contains(e.target as Node)) {
```

Comment:
TypeScript should guarantee that `containerRef` is a `RefObject` here.

```suggestion
  if (containerRef.current?.contains(e.target as Node)) {
```